src/Native/Runtime/unix/UnixContext.cpp

```diff
@@ -361,10 +366,23 @@ void UnwindCursorToRegDisplay(unw_cursor_t *cursor, unw_context_t *unwContext, R
     ASSIGN_REG(R14, R14)     \
     ASSIGN_REG(R15, R15)
 
-#define ASSIGN_TWO_ARGUMENT_REGS(arg0Reg, arg1Reg) \
-    MCREG_Rdi(nativeContext->uc_mcontext) = arg0Reg;      \
+#define ASSIGN_TWO_ARGUMENT_REGS(arg0Reg, arg1Reg)    \
+    MCREG_Rdi(nativeContext->uc_mcontext) = arg0Reg;  \
     MCREG_Rsi(nativeContext->uc_mcontext) = arg1Reg;
 
+#elif defined(_X86_)
+#define ASSIGN_CONTROL_REGS \
+    ASSIGN_REG(Eip, IP)     \
+    ASSIGN_REG(Esp, Rsp)
+
+#define ASSIGN_INTEGER_REGS  \
+    ASSIGN_REG(Ebx, Rbx)     \
+    ASSIGN_REG(Ebp, Rbp)
+
+#define ASSIGN_TWO_ARGUMENT_REGS(arg0Reg, arg1Reg)    \
+    MCREG_Edi(nativeContext->uc_mcontext) = arg0Reg;  \
```

Comment:
These should be ecx and edx unless we are going to use calling convention different from what we use in coreclr and on windows.